modules/apps/27-interchain-accounts/types/packet.go

```diff
@@ -1,9 +1,40 @@
 package types
 
 import (
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-func (packet InterchainAccountPacketData) GetBytes() []byte {
-	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&packet))
+// ValidateBasic performs basic validation of the interchain account packet data.
+// The memo may be empty.
+func (iapd InterchainAccountPacketData) ValidateBasic() error {
+	if iapd.Data == nil {
+		return sdkerrors.Wrap(ErrInvalidOutgoingData, "packet data cannot be empty")
+	}
+	// TODO: add type validation when data type enum supports unspecified type
+
+	return nil
+}
+
+// GetBytes returns the JSON marshalled interchain account packet data.
+func (iapd InterchainAccountPacketData) GetBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&iapd))
+}
+
+// GetBytes returns the JSON marshalled interchain account CosmosTx.
+func (ct CosmosTx) GetBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&ct))
+}
+
+// UnpackInterfaces implements UnpackInterfacesMessage.UnpackInterfaces
+func (ct CosmosTx) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
```

Comment:
If you type contains an Any or a struct that contains an Any, this function must be implemented or JSON marshal/unmarshal will fail (the compiler won't catch this, requires a test)